src/Stryker.Core/Stryker.Core/ProjectComponents/SimpleFileLeaf.cs

```diff
@@ -0,0 +1,39 @@
+using DotNet.Globbing;
+using Stryker.Core.Mutants;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stryker.Core.ProjectComponents
+{
+    public class SimpleFileLeaf: ProjectComponent, IReadOnlyFileLeaf
```

Comment:
This is really needed only GitDiffProvider. Although it's useful in tests also.

Now, I guess this could be somehow inserted in the tree of abstractions here since there is some logic duplication with ExcludableProjectComponent. But my brain already hurts from abstractions here and the duplication is rather small anyway.

Reply:
I'm not sure what this class is doing.. Is this a non-language-specific file? Should this be an abstract class?